src/dotnet/ToolManifest/ToolManifestFinder.cs

```diff
@@ -28,92 +31,143 @@ public ToolManifestFinder(DirectoryPath probStart, IFileSystem fileSystem = null
-                        {
-                            MissingMemberHandling = MissingMemberHandling.Ignore
-                        });
+                    findAnyManifest = true;
+                    SerializableLocalToolsManifest deserializedManifest =
+                        DeserializableLocalToolsManifest(possibleManifest);
 
-                    var errors = new List<string>();
-
-                    if (!deserializedManifest.isRoot)
+                    foreach (ToolManifestPackage p in GetToolManifestPackageFromOneManifestFile(deserializedManifest, possibleManifest))
                     {
-                        errors.Add("isRoot is false is not supported.");
+                        if (!result.Any(addedToolManifestPackages =>
+                            addedToolManifestPackages.PackageId.Equals(p.PackageId)))
+                        {
+                            result.Add(p);
+                        }
                     }
 
-                    if (deserializedManifest.version != 1)
+                    if (deserializedManifest.isRoot)
                     {
-                        errors.Add(string.Format("Tools manifest format version {0} is not supported.",
-                            deserializedManifest.version));
+                        return result;
                     }
+                }
+            }
 
-                    foreach (var tools in deserializedManifest.tools)
-                    {
-                        var packageLevelErrors = new List<string>();
-                        var packageIdString = tools.Key;
-                        var packageId = new PackageId(packageIdString);
+            if (!findAnyManifest)
+            {
+                throw new ToolManifestCannotFindException(
+                    string.Format(LocalizableStrings.CannotFindAnyManifestsFileSearched,
+                        string.Join(Environment.NewLine, allPossibleManifests.Select(f => f.Value))));
+            }
 
-                        string versionString = tools.Value.version;
-                        NuGetVersion version = null;
-                        if (versionString is null)
-                        {
-                            packageLevelErrors.Add(LocalizableStrings.MissingVersion);
-                        }
-                        else
-                        {
-                            var versionParseResult = NuGetVersion.TryParse(
-                                versionString, out version);
+            return result;
+        }
 
-                            if (!versionParseResult)
-                            {
-                                packageLevelErrors.Add(string.Format(LocalizableStrings.VersionIsInvalid, versionString));
-                            }
-                        }
+        private SerializableLocalToolsManifest DeserializableLocalToolsManifest(FilePath possibleManifest)
```

Comment:
Should this method name be DeserializeLocalToolsManifest?

Reply:
fixed, renamed as suggested